inc/mailcollector.class.php

```diff
@@ -911,33 +911,51 @@ function buildTicket($uid, $options = []) {
 
             $begin_strip     = -1;
             $end_strip       = -1;
-            $begin_match     = "/".NotificationTargetTicket::HEADERTAG.".*".
-                                 NotificationTargetTicket::HEADERTAG."/";
-            $end_match       = "/".NotificationTargetTicket::FOOTERTAG.".*".
-                                 NotificationTargetTicket::FOOTERTAG."/";
+            $header_tag      = NotificationTargetTicket::HEADERTAG;
+            $header_pattern  = $header_tag . '.*' . $header_tag;
+            $footer_tag      = NotificationTargetTicket::FOOTERTAG;
+            $footer_pattern  = $footer_tag . '.*' . $footer_tag;
             foreach ($content as $ID => $val) {
                // Get first tag for begin
                if ($begin_strip < 0) {
-                  if (preg_match($begin_match, $val)) {
+                  if (preg_match('/' . $header_pattern . '/', $val)) {
                      $begin_strip = $ID;
                   }
                }
                // Get last tag for end
                if ($begin_strip >= 0) {
-                  if (preg_match($end_match, $val)) {
+                  if (preg_match('/' . $footer_pattern . '/', $val)) {
                      $end_strip = $ID;
                      continue;
                   }
                }
             }
 
-            if ($begin_strip >= 0) {
-               // Clean first and last lines
-               $content[$begin_strip] = preg_replace($begin_match, '', $content[$begin_strip]);
-            }
-            if ($end_strip >= 0) {
-               // Clean first and last lines
-               $content[$end_strip] = preg_replace($end_match, '', $content[$end_strip]);
+            if ($begin_strip >= 0 && $end_strip >= 0 && $begin_strip === $end_strip) {
```

Comment:
This part fixes the stripping of mails received as "compact" html (compact = all html on same line).